### PR TITLE
apache-ant*: obsolete apache-ant-1.9

### PR DESCRIPTION
**Pull request by McDaMastR**
#### Description

There are currently two ports for [Apache Ant](https://ant.apache.org): `apache-ant-1.9` and `apache-ant`, which follow the 1.9.x and 1.10.x release series, respectively.

On 2024-06-19 (about two years ago), the 1.9.x series was officially EOL'd (see the [announcement](https://lists.apache.org/thread/f6jw4v3gjwhqt5fz25og0my2o6xwvvm1)).  While the 1.10.x series is still maintained, with the latest release being on 2026-04-10 (see the [download page](https://ant.apache.org/bindownload.cgi)).

So given that `apache-ant-1.9` currently has no dependents, this pull request obsoletes `apache-ant-1.9` in favour of `apache-ant`.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 26.6.2 25G83 arm64
Command Line Tools 26.6.0.0.1781586589

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?

### CODE REVIEW

File: devel/apache-ant-1.9/Portfile [hunk 1]
```diff
@@ -1,69 +1,16 @@
...
-    }
-
-    ln -s ../share/java/apache-ant/bin/ant ${destroot}${prefix}/bin/ant
-}
-
-livecheck.type          regex
-livecheck.url           https://www.apache.org/dist/ant/binaries/
-livecheck.regex         apache-ant-([quotemeta ${branch}](?:\\.\\d+)*)-bin\\.tar\\.bz2
+replaced_by             apache-ant
+obsolete.note           The Apache Ant 1.9.x series is EOL.
```
Comment (reneeotten) [inline devel/apache-ant-1.9/Portfile:17]: please add a note when the port can be removed from MacPorts (i.e., one year from now).

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by barracuda156** [comment]
@McDaMastR Did you confirm that `apache-ant` builds and works on all supported by MacPorts systems? (Sufficient to confirm 10.6 x86_64.)

P. S. It is somewhat misleading to claim “no dependents”, since these ports are, at least presumably, drop-ins. `apache-ant-1.9` has as many dependents as `apache-ant` does.

File: devel/apache-ant-1.9/Portfile (hunk 1 above)
Comment (McDaMastR) [reply devel/apache-ant-1.9/Portfile:17]: Done.

**Comment by McDaMastR** [comment]
@barracuda156

> Did you confirm that `apache-ant` builds and works on all supported by MacPorts systems? (Sufficient to confirm 10.6 x86_64.)

Unfortunately I only have access to my one MacBook (running macOS Tahoe), so I'm unable to test on other supported macOS systems.  That said, if it is found that `apache-ant` fails to build on a supported macOS system, then I'd think the best course of action would be to instead deprecate the port.  That way users who need the 1.9.x series can use it, and users who don't are informed of the EOL status and encouraged to use the maintained 1.10.x series.

But ideally, all users should be migrated away from an EOL release series to avoid potential unaddressed security vulnerabilities.  So unless we have reason to believe `apache-ant` does not build on a supported system, I'd say obsoleting the port is the right thing to do.

> P. S. It is somewhat misleading to claim “no dependents”, since these ports are, at least presumably, drop-ins. `apache-ant-1.9` has as many dependents as `apache-ant` does.

I was just going by the port's [details webpage](https://ports.macports.org/port/apache-ant-1.9/details), which doesn't list any dependents.  Though you're right that in many cases, both `apache-ant` and `apache-ant-1.9` likely could work as a port's dependency, so `apache-ant-1.9` does kind of have dependents.  It just doesn't have any that explicitly specify a need for the 1.9.x series.

**Comment by barracuda156** [comment]
It kinda feels odd when users who do not use the port and cannot even test it decide for those who do. In economics it is called "externalizing costs". A bad way of decision making. If the port is not selected by default, what is even the problem to have it? MacPorts carries multiple versions of a lot of ports, many of those old. This is not something that goes against conventional practice.

**Comment by McDaMastR** [comment]
(No, I did not use AI to write this.  All words from me are typed by me.)

> It kinda feels odd when users who do not use the port and cannot even test it decide for those who do.

First of all, this is just a pull request.  I'm simply proposing a change and welcoming discussion on it.  I'm not deciding anything for anyone.

Second, I'm unsure what exactly you're implying.  Are you saying you must use a port to propose changes to it?  I don't think that's a reasonable requirement.  I've recently been going around updating a bunch of java-related ports because I want to help fix this large [trac ticket](https://trac.macports.org/ticket/60206) I found.  Should I not be doing that because I don't use any of those ports?

I also want to note that I don't think I should be expected to be able to test a change on every supported system.  I'm not rich enough to own that many computers.  I also don't think I should be expected to be able to test on the minimum/oldest supported system just to propose a change.  I highly doubt most contributors to MacPorts have access to an x86_64 computer with macOS 10.6 installed.  And I think that's fine.  Those who do can help those who don't.

> In economics it is called "externalizing costs". A bad way of decision making.

I'm certainly no economist, but from [what I can tell](https://sustainability-directory.com/fundamentals/externalized-costs-fundamentals), externalising costs has to do with shifting the cost of production onto a third party via non-directly-monetary means, commonly a decrease in quality of life.  I'm afraid I can't tell how that relates or is analogous to our situation, given there's no obvious cost of production here (besides time, I suppose).

> If the port is not selected by default, what is even the problem to have it? MacPorts carries multiple versions of a lot of ports, many of those old.

By this logic, every major release of a project should get its own port and old ports should never be obsoleted or removed.  But this is a bad idea.  More ports means more work for maintainers, which if not performed, leads to rotting unmaintained ports which break on users.

> This is not something that goes against conventional practice.

To my understanding, MacPorts has no official policy on obsoleting or removing ports, but tends to remove ports that (a) package unmaintained software, (b) provide negligible value or redundant functionality, and (c) have a minimal or non-existent userbase.  AFAICT, `apache-ant-1.9` fits these three criteria.  Apache Ant 1.9.x is EOL, provides the same essential functionality as Apache Ant 1.10.x (packaged by `apache-ant`), and (according to the port's [details webpage](https://ports.macports.org/port/apache-ant-1.9/details)) has had only one installation in the past year.

**Comment by barracuda156** [comment]
You suggested to outright remove the port, on the grounds that you think nobody needs it and there is a newer version which is a drop-in (besides, installation statistics means not much generally, since it is an opt-in, and less so for < 10.8 or about that, were it won’t work at all without manual trickery). My suggestion is to confirm first whether this change does not break stuff for users, i.e. that at least the second assumption holds. No one asks to add a port for every version of everything. But when such ports are added, usually there is some credible enough reason. Also hardly the port added speculatively: most likely it was to address a specific issue.

I don’t decide anything here either, but anyone can review the proposal. (For the record, this is not about supporting powerpc at all, that is not invoked as an argument from my side.)

**Comment by McDaMastR** [comment]
> You suggested to outright remove the port, on the grounds that you think nobody needs it and there is a newer version which is a drop-in

Well, moreso on grounds of wanting to avoid users needlessly/unknowingly using insecure software.

> (besides, installation statistics means not much generally, since it is an opt-in, and less so for < 10.8 or about that, were it won’t work at all without manual trickery).

I wasn't aware about those issues with < ~10.8.  In that case, it would be fair to be sceptical of the installation stats for those older systems.

We could instead look at the number of trac tickets opened for `apache-ant-1.9`, of which there are two in the port's history.  Namely [57013](https://trac.macports.org/ticket/57013) and [58787](https://trac.macports.org/ticket/58787), opened eight and seven years ago, respectively.  This small number of tickets could suggest a minimal userbase, and their age suggests recent usage of the port is even less.

But of course, ticket numbers is a tenuous measure of user count.  Do you know of a more reliable way to query how many users of this port there are?

> My suggestion is to confirm first whether this change does not break stuff for users, i.e. that at least the second assumption holds. No one asks to add a port for every version of everything. But when such ports are added, usually there is some credible enough reason. Also hardly the port added speculatively: most likely it was to address a specific issue.

Indeed, the port was added for a specific reason.  Judging by trac ticket [53555](https://trac.macports.org/ticket/53555) and its accompanying [pull request](https://github.com/macports/macports-ports/pull/331) (merged ~8.5 years ago), `apache-ant-1.9` was added to allow macOS 10.4-10.6 users to use the natively shipped Apple Java program with Apache Ant.  Apple shipped Java 5 on macOS 10.4 and Java 6 on macOS 10.6 (I believe Apple stopped shipping Java from macOS 10.7).  Meanwhile, Apache Ant 1.9.x requires Java 5+ and Apache Ant 1.10.x requires Java 8+.

Today, Java 5-7 are EOL; at least Java 8 should be used if possible.  So I think the main reason to not obsolete `apache-ant-1.9` would be if macOS 10.6 users must use the Apple-shipped Java 6 (i.e. are unable to install Java 8/11/...).  In which case, I'd agree obsoleting the port is not the right move and would instead argue for deprecating it.
